content/client-lib-development-guide/features.textile

```diff
@@ -1697,13 +1735,40 @@ class RealtimeChannel:
   unsubscribe(String, (Message) ->) // RTL8a
   setOptions(options: ChannelOptions) => io // RTL16
 
+class BatchOperations:
+  publish([BatchSpec]) => BatchResult<BatchPublishResponse> // BO2a
```

Comment:
Thinking about partial success, I think I'd like to get back something which helps me understand which of the things I tried to publish failed.

Whilst I understand why the REST API is designed as it is, I think there's an opportunity to wrap it in a client-side abstraction which might be more useful.

For example, what if it looked like this:

```
batch := client.newBatchPublish()
req1 := batch.publish("channel1", "msg1")
req2 := batch.publish("channel1", "msg2")
req3 := batch.publish("channel2", "msg3")
results, err := batch.send()
```

and `results` somehow helps me correlate any errors with one or more of `req1`, `req2`, or `req3`.